web_external/js/views/widgets/AddSourceWidget.js

```diff
@@ -6,39 +6,63 @@ minerva.views.AddSourceWidget = minerva.View.extend({
         'submit #m-add-source-form': function (e) {
             e.preventDefault();
 
+            // TODO better to have single buttons instead of radio?
+            // That way user only has to click once instead of radio + Add
```

Comment:
I agree, in fact I don't even think this necessarily has to be a widget. Rather, the + icon could just trigger a dropdown of clickable source types.

Simplifying/changing that can be it's own issue though.